pkg/env/action.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package env
+
+import (
+	"context"
+
+	"sigs.k8s.io/e2e-framework/pkg/internal/types"
+)
+
+const(
+	roleSetup = iota
+	roleBefore
+	roleAfter
+	roleFinish
```

Comment:
I'm a little unclear on how `roleSetup` is different from `roleBefore`, likewise for `roleAfter` and `roleFinish`. Perhaps we should have explicit documentation about this somewhere?

Reply:
Fair. There are four roles an action can have
* Setup - executed at start of a suite (package TestMain)
* Before - executed prior each env.Test()
* After - executed after each test
* Finish - Executed at the end of a test suite (prior to exiting TestMain)